tests/core/test_test.py

```diff
@@ -496,3 +496,33 @@ def test_test_generation(tmp_path: Path) -> None:
     test = load_yaml(context.path / c.TESTS / "foo/bar.yaml")
     assert len(test) == 1
     assert "new_name" in test
+
+
+def test_source_func() -> None:
+    result = SqlModelTest(
```

Comment:
fyi could also use the `_create_test` helper if you wanted to refactor this I think

Reply:
i guess check out the other test methods, iirc there's a pattern to how these are tested with the helpers